queue_job/models/queue_job.py

```diff
@@ -14,9 +14,18 @@
 
 _logger = logging.getLogger(__name__)
 
+from odoo.models import MetaModel
+
 
 def channel_func_name(method):
-    return '<%s>.%s' % (method.im_class._name, method.__name__)
+    method_class = False
+    for key, value in method.func_globals.iteritems():
```

Comment:
I saw your commit's comment "im_class seems not accessible". What is the type of `method` in your use case?